### PR TITLE
feat: add ora submission view filter for turnitin integration

### PR DESCRIPTION
**Pull request by BryanttV**
### Description
This PR adds a pipeline step that modifies the ORA submission view rendering process. This step changes the default template by a custom template that has a warning message that notifies the student the Turnitin plagiarism check is enabled.

### How to Test

1. Install this plugin with this branch in the LMS/CMS services.
2. Install `openedx-filters` in the LMS/CMS services with [these changes](https://github.com/eduNEXT/openedx-filters/pull/1)
3. Install `edx-ora2` in the LMS/CMS services with [these changes](https://github.com/eduNEXT/edx-ora2/pull/13)
4. Add the Open edX Filters configuration to your environment. You can use this inline plugin:

    ```yaml
    name: filters-settings
    version: 0.1.0
    patches:
      openedx-common-settings: |
        OPEN_EDX_FILTERS_CONFIG = {
          "org.openedx.learning.ora.submission_view.render.started.v1": {
            "fail_silently": False,
            "pipeline": [
              "platform_plugin_turnitin.extensions.filters.ORASubmissionViewTurnitinWarning",
            ]
          },
        }
    ```
5. Restart services
6. In a unit of the course create a new ora assessment.
7. In the LMS, in the submission step (**1. Your Response**), you'll see:

    image

### CODE REVIEW

File: platform_plugin_turnitin/extensions/filters.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"""Filters for the Turnitin plugin."""
+
+from openedx_filters import PipelineStep
+
+
+class CheckTurnitinForPlagiarism(PipelineStep):
+    """Check if Turnitin plagiarism check is enabled."""
```
Comment (mariajgrimaldi) [inline platform_plugin_turnitin/extensions/filters.py]: is this an accurate description of the class?

Comment (BryanttV) [reply platform_plugin_turnitin/extensions/filters.py]: Maybe not, I have updated the filter name and description.

File: platform_plugin_turnitin/templates/turnitin/oa_response.html [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+                </div>
+
+                <div class="openassessment__message message">
+                  <div class="message__content">
+                    <p>
+                      <span class="icon fa fa-exclamation-triangle"></span>
+                      {% blocktrans %}
+                        The response text and files related to your response will be sent to Turnitin for a plagiarism check.
+                        Please read <a href="https://www.turnitin.com/agreement.asp" target="_blank">Turnitin's EULA</a>
+                        before submitting, as by doing so you are explicitly accepting to the agreement.
```
Comment (mariajgrimaldi) [inline platform_plugin_turnitin/templates/turnitin/oa_response.html]:
```suggestion
                        Texts and files related to your submission will be sent to Turnitin for a plagiarism check.
                        Please read <a href="https://www.turnitin.com/agreement.asp" target="_blank">Turnitin's EULA</a>
                        before submitting, as by doing so, you are explicitly accepting the EULA agreement.
```

File: requirements/base.in [hunk 3]
```diff
@@ -7,5 +7,5 @@ djangorestframework     # RESTful API framework
 edx-drf-extensions      # Extensions to Django REST Framework for edX
 celery                  # Asynchronous task queue
 edx-submissions         # Submissions API for edX
-attrs                   # Classes without boilerplate
 openedx-events @ git+https://github.com/edunext/openedx-events.git@9.4.0/edues # Open edX event tracking
+openedx-filters @ git+https://github.com/edunext/openedx-filters.git@bav/ora-student-view-filter # Open edX filters
```
Comment (mariajgrimaldi) [inline requirements/base.in]: Let's put a pin on this so we can solve it later: who should manage this kind of dependency? Us while building using tutor patches or the plugin?

**Review by mariajgrimaldi — APPROVED** [review]

File: platform_plugin_turnitin/templates/turnitin/oa_response.html (hunk 2 above)
Comment (BryanttV) [reply platform_plugin_turnitin/templates/turnitin/oa_response.html]: Done

**Pull request merged**